src/components/Button.jsx

```diff
@@ -0,0 +1,54 @@
+import PropTypes from "prop-types";
+import { Link } from "react-router-dom";
+import classNames from "classnames";
+
+const Button = ({ text, variant, size, to, fullWidth, children, ...props }) => {
+  const buttonClass = classNames(
+    "inline-flex gap-2 items-center justify-center font-semibold transition-colors duration-300 rounded-lg",
+    {
+      "bg-orange-500 text-white hover:bg-red-500 justify-between":
+        variant === "default",
+      "bg-black text-white hover:bg-orange-500": variant === "black",
+      "border-2 border-orange-500 !text-xl text-orange-500 hover:bg-black hover:border-black hover:text-white group":
+        variant === "outline",
+      "bg-black text-white hover:bg-orange-500 !rounded-full !gap-0 !p-4 w-[60px] h-[60px]":
+        variant === "circular",
+      "w-full !justify-center": fullWidth,
+      "py-4 px-10 text-xl": size === "large",
+      "py-2 px-6 text-lg": size === "medium",
+      "py-1 px-4 text-sm": size === "small",
+    }
+  );
+
+  const ButtonComponent = to ? Link : "button";
```

Comment:
Lets not use `ButtonComponent` because in Link we use `to` and in `button` we use `onClick`. So its better to do some repetition here in favour of code readability.
Like 

return (
   { to  ? ( <Link .....>) : <button ....>
   }
)

Reply:
Hi, @rt4914 applied according to your comment. Thanks.